### ISSUE TITLE
Android Exoplayer - Failed

### ISSUE THREAD

**Issue by MohibWasay** [TheWidlarzGroup/react-native-video#2631]
## Android ExoPlayer - Build Execution Fails

I am trying to run Android Build with ExoPlayer and I am getting the following error.

````
error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
/Users/mohibwasay/Desktop/workspace/relevnt/relevnt-app/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:330: error: cannot find symbol
        ImageButton playButton = playerControlView.findViewById(R.id.exo_play);
                                                                    ^
  symbol:   variable exo_play
  location: class id
/Users/mohibwasay/Desktop/workspace/relevnt/relevnt-app/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:342: error: cannot find symbol
        ImageButton pauseButton = playerControlView.findViewById(R.id.exo_pause);
                                                                     ^
````

#### build.gradle

````
plugins {
    id "com.android.application" version '7.1.2' apply false
    id "com.android.library" version '7.1.2' apply false
    id "org.jetbrains.kotlin.android" version "1.6.10" apply false
    id "com.google.gms.google-services" version "4.3.10" apply false
    id "com.osacky.doctor" version "0.8.0"
}

ext {
    buildToolsVersion = "33.0.0-rc2"
    minSdkVersion = 21
    compileSdkVersion = 31
    targetSdkVersion = 30
    supportLibVersion = "28.0.0"
    firebaseBomVersion = "28.2.0"
    googlePlayServicesVersion = "18.+"
    ndkVersion = "21.4.7075529"
    googlePlayServicesAuthVersion = "19.2.0" 
    kotlin_version = "1.6.10"
}


task clean(type: Delete) {
    delete rootProject.buildDir
}

````

**Comment by MohibWasay** [comment]
Any updates on this one.

**Comment by hueniverse** [comment]
Have you tried to compare your setup to https://github.com/react-native-video/react-native-video-test or one of the examples and see if you are able to get those working?

**Comment by MohibWasay** [comment]
@hueniverse . The difference is that I am having Kotlin based android app.

**Comment by freeboub** [comment]
I don't think kotlin is the root cause. I already had similar issue, it was linked to buildTool (if my memory is still Ok). Did you try to change buildTool version ? 
Notice that last buildTool version is 30.0.0 - rc4: https://developer.android.com/studio/releases/build-tools
how can you use :     buildToolsVersion = "33.0.0-rc2" ?

**Comment by darnmason** [comment]
Can anyone confirm what was the cause of this issue?

**Comment by androideveloper** [comment]
This should be fixed to make rn-video compatible with next version of AGP plugin
https://medium.com/androiddevelopers/5-ways-to-prepare-your-app-build-for-android-studio-flamingo-release-da34616bb946

You can reproduce it if you enable this flag in gradle.properties
-android.nonTransitiveRClass=true

**Comment by MobolajiRichard** [comment]
I am getting this same issue, anyone fixed it yet...

React Native - 0.73
React Native Video - 5.2.1

I can't seem to find any solution. Please help

**Comment by s43** [comment]
@MobolajiRichard forget about that, I spent two days stuck with this problem and finally I found that there's a lack of documentation, it seems that since v6 Exo Player is being used by default, and the path we are asked to put within the react-native.config.js does no longer exist..

**Comment by PrasadDevadigaPhilips** [comment]
We are also facing the same issue when we update the React native version to 0.73
Is there any fix available on top of 5.2.1 ?

**Comment by prasaddevadiga** [comment]
We are also facing the same issue when we update the React native version to 0.73
Is there any fix available on top of 5.2.1 ?

**Comment by CZDevops** [comment]
> We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?

I change this line myself to:

ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play);
ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);

**Comment by ICGVIP** [comment]
> > We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?
> 
> I change this line myself to:
> 
> ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play); ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);

This works but is there a better solution to avoid touching the code itself ?

**Comment by prasaddevadiga** [comment]
> > We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?
> 
> I change this line myself to:
> 
> ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play); ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);

Thanks for the response, this we tried and it worked but I was looking for patch release.

**Comment by CZDevops** [comment]
> > > We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?
> > 
> > 
> > I change this line myself to:
> > ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play); ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);
> 
> Thanks for the response, this we tried and it worked but I was looking for patch release.

I've not tried it in RN, but can you use the package: https://www.npmjs.com/package/patch-package to create your own patch till an official patch is released?

**Comment by amandaciobanu** [comment]
@CZDevops Where did you modify these lines?I am having the same issues and could not figure out where you applied the changes 

Thank you very much

**Comment by CZDevops** [comment]
> com.google.android.exoplayer2.R.id.exo_play

node_modules\react-native-video\android-exoplayer\src\main\java\com\brentvatne\exoplayer\ReactExoplayerView.java

**Comment by CZDevops** [comment]
> @CZDevops Where did you modify these lines?I am having the same issues and could not figure out where you applied the changes
> 
> Thank you very much

node_modules\react-native-video\android-exoplayer\src\main\java\com\brentvatne\exoplayer\ReactExoplayerView.java

**Comment by rajkamal-360** [comment]
> > We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?
> 
> I change this line myself to:
> 
> ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play); ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);

how can i change the library code.

**Comment by oem2** [comment]
> > We are also facing the same issue when we update the React native version to 0.73 Is there any fix available on top of 5.2.1 ?
> 
> I change this line myself to:
> 
> ImageButton playButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_play); ImageButton pauseButton = playerControlView.findViewById(com.google.android.exoplayer2.R.id.exo_pause);

This solved it for me. Thanks so much.
